### PR TITLE
spring

### PR DESCRIPTION
**Pull request by pavlokhomiak**

### CODE REVIEW

**Review by fyarkiy — APPROVED** [review]
Great work! check to unnecessary empty line in pom.xml

File: src/main/java/dao/impl/UserDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import model.User;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Primary;
+import org.springframework.stereotype.Repository;
+
+@Primary
```
Comment (RomanDubovskyi) [inline src/main/java/dao/impl/UserDaoImpl.java]: what does Primary do?

File: src/main/java/dao/impl/UserDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+@Repository
+public class UserDaoImpl implements UserDao {
+
+    @Autowired
+    private SessionFactory sessionFactory;
```
Comment (RomanDubovskyi) [inline src/main/java/dao/impl/UserDaoImpl.java:15]: it's better to use injection through constructor

File: src/main/java/service/impl/UserServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import service.UserService;
+
+@Service
+public class UserServiceImpl implements UserService {
+
+    @Autowired
+    @Qualifier(value = "userDaoImpl")
+    private UserDao userDao;
```
Comment (RomanDubovskyi) [inline src/main/java/service/impl/UserServiceImpl.java:12]: same here

File: src/main/java/service/impl/UserServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import service.UserService;
+
+@Service
+public class UserServiceImpl implements UserService {
+
+    @Autowired
+    @Qualifier(value = "userDaoImpl")
```
Comment (RomanDubovskyi) [inline src/main/java/service/impl/UserServiceImpl.java]: do we need qualifier?

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**